src/Common/UInt128.h

```diff
@@ -48,10 +48,9 @@ struct UInt128
 
     String toHexString() const
     {
-        std::ostringstream os;
-        os.exceptions(std::ios::failbit);
-        os << std::setw(16) << std::setfill('0') << std::hex << high << low;
-        return String(os.str());
+        String res(2 * sizeof(UInt128), 0);
+        writeHexUIntLowercase(*this, res.data());
```

Comment:
Looks like we get `10000000000000000` instead of `0000000000000001` now?

Reply:
No, [it](https://github.com/ClickHouse/ClickHouse/blob/06446b4f08a142d6f1bc30664c47ded88ab51782/src/Common/hex.h#L57) takes number in little endian and prints it in big endian